SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityStartup.java

```diff
@@ -123,7 +124,13 @@ void checkSchedulerForInconsistentState() {
         case ACTIVE:
         case SYSTEM_COOLDOWN:
         case DEPLOYING_TO_UNPAUSE:
-          checkActiveRequest(requestWithState, deployKeyToPendingTaskId, now);
+          if(checkActiveRequest(requestWithState, deployKeyToPendingTaskId, now)) {
+            numPendingStartupRequests++;
+            if (numPendingStartupRequests > THRESHOLD) {
+              LOG.info("Reached startup task threshold of %s", THRESHOLD);
+              return; // todo: abort Singularity??
```

Comment:
To abort we'll use the `SingularityAbort` similar to the usage [here](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerDelegator.java#L98). You can add a new `AbortReason` to the enum